attr.c

```diff
@@ -744,6 +745,19 @@ static struct attr_stack *read_attr_from_index(struct index_state *istate,
 	if (!istate)
 		return NULL;
 
+	/*
+	 * In the case of cone-mode sparse-checkout, getting the
+	 * .gitattributes file from a directory is meaningless: all
+	 * contained paths will be sparse if the .gitattributes is also
+	 * sparse. In the case of a sparse index, it is critical that we
+	 * don't go looking for one as it will expand the index.
+	 */
+	init_sparse_checkout_patterns(istate);
```

Comment:
I think this is unnecessary, as [`path_in_sparse_checkout()`](https://github.com/microsoft/git/pull/410/files#diff-13026f61c17631884dc4c6ee9128710ef8801844114eaedaf0a13db649b4b0a2R1513-R1529) already calls that function.